var/spack/repos/builtin/packages/hiop/package.py

```diff
@@ -65,15 +65,16 @@ class Hiop(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on('magma+cuda', when='+cuda')
     depends_on('magma+rocm', when='+rocm')
-    depends_on('magma@2.5.4:', when='@0.4:+cuda')
-    depends_on('magma@2.6.1:', when='@0.4.6:+cuda')
-    depends_on('magma@2.5.4:', when='@0.4:+rocm')
-    depends_on('magma@2.6.1:', when='@0.4.6:+rocm')
-
-    # MAGMA v2.6.2 was not released when v0.5.4 was released, but we need
-    # commits from master branch. The most recent tested commit is
-    # 5959b8783e45f1809812ed96ae762f38ee701972
-    depends_on('magma@master', when='@0.5.4:')
+
+    # Depends on Magma when +rocm or +cuda
+    magma_ver_constraints = (
+        ('2.5.4', '0.4'),
+        ('2.6.1', '0.4.6'),
+        ('2.6.2rc1', '0.5.4'),
```

Comment:
```suggestion
        ('2.6.2', '0.5.4'),
```
Spack actually sorts 2.6.2rc1 as being newer than 2.6.2, so if you want to be able to use 2.6.2 when it's fully released you'll need to change this

Reply:
Hm, that's in interesting way for spack to sort versions 😅 . What do you think about `2.6.1.99` so it doesn't conflict with 2.6.2 when it's released? (CCing magma maintainers, seems like spackbot doesn't pick them up if the package is added later on @G-Ragghianti @luszczek @stomov)

Reply:
> so it doesn't conflict with 2.6.2 when it's released?

What do you mean?

Reply:
Oh, I see. depending on `2.6.2:` is better than `2.6.2rc1:`. Thanks!